File: backend/app/routers/admin_analytics.py
from datetime import UTC, datetime, timedelta

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db
from app.core.permissions import require_permission
from app.models.user import User
from app.schemas.admin_analytics import AnalyticsResponse
from app.services import admin_analytics_service

router = APIRouter(prefix="/admin/analytics", tags=["admin-analytics"])


@router.get("", response_model=AnalyticsResponse)
async def get_analytics(
    date_from: datetime | None = None,
    date_to: datetime | None = None,
    current_user: User = Depends(require_permission("View Analytics")),
    db: AsyncSession = Depends(get_db),
):
    resolved_to = date_to or datetime.now(UTC)
    resolved_from = date_from or (resolved_to - timedelta(days=365))
    return await admin_analytics_service.get_analytics(db, date_from=resolved_from, date_to=resolved_to)

File: frontend/src/lib/api/admin-analytics.ts
import { apiFetch } from "./client";
import type { AnalyticsResponse } from "@/lib/types/admin-analytics";

export async function getAnalytics(params?: {
  date_from?: string;
  date_to?: string;
}): Promise<AnalyticsResponse> {
  const qs = new URLSearchParams();
  if (params?.date_from) qs.set("date_from", params.date_from);
  if (params?.date_to) qs.set("date_to", params.date_to);
  const suffix = qs.toString() ? `?${qs.toString()}` : "";
  return apiFetch<AnalyticsResponse>(`/admin/analytics${suffix}`);
}
